LoopBack/LBFile.m

```diff
@@ -40,15 +40,19 @@ - (void)uploadWithSuccess:(LBFileUploadSuccessBlock)success
                                                                           fileName:self.name
                                                                        contentType:mimeType
                                                                             length:length];
-
-   [self invokeMethod:@"upload"
-           parameters:@{ @"container": self.container,
-                         @"name": self.name,
-                         @"file": streamParam }
-              success:^(id value) {
-                  success();
-              }
-              failure:failure];
+    
+    [self invokeMethod:@"upload"
+            parameters:@{ @"container": self.container,
+                          @"name": self.name,
+                          @"file": streamParam }
+               success:^(id value) {
+                   NSAssert([[value class] isSubclassOfClass:[NSDictionary class]],
+                            @"Received non-Dictionary: %@", value);
+                   NSDictionary *fileDictionary = [(NSDictionary*)value valueForKeyPath:@"result.files.file"][0];
```

Comment:
How about adding asserting to check `fileDictionary` being non `nil` here?  Reading the component storage code, the key path `result.files.file` should always be fine, but it was not obvious in [the API doc](https://apidocs.strongloop.com/loopback-component-storage/#storageservice-prototype-upload) to me, so I thought it is better to be a bit protective here.